src/mqtt/MQTT.cpp

```diff
@@ -559,21 +555,25 @@ void MQTT::onSend(const meshtastic_MeshPacket &mp, const meshtastic_MeshPacket &
             return;
         }
     }
-    if (mp.pki_encrypted || ch.settings.uplink_enabled) {
-        const char *channelId = mp.pki_encrypted ? "PKI" : channels.getGlobalId(chIndex);
+    // Either encrypted packet (we couldn't decrypt) is marked as pki_encrypted, or we could decode the PKI encrypted packet
+    bool isPKIEncrypted = mp_encrypted.pki_encrypted || mp_decoded.pki_encrypted;
```

Comment:
The comment on line 540 seems inconsistent with the comment down here and I'm not sure which is right -- if we get a PKI-encrypted packet that we _can_ decode (i.e., to us), should it hit the branch above? I guess it looks like it'd fall through both of the branches within the above, so I suppose it probably needn't bother.

Reply:
Ah, I see what you mean. It's only not decoded when it's PKI encrypted and not to us, will update the wording there.

Reply:
Ah, I think I see my confusion here, problem on my end. I thought that line was saying "if it was ever PKI encrypted, it will not have been decoded", which isn't true if it was succesfully decoded (as this line shows). Looking again, I guess what's meant is more "if this is a PKI message we can't decrypt, the name 'mp_decoded' is slightly misleading because it will not actually be decoded, therefore we need to guard against that".

So yeah, issue with understanding on my end. Carry on :)

Reply:
Haha, yes, that was indeed the intention behind that comment.